docs/Overview.md

```diff
@@ -414,11 +413,11 @@ def __call__(self, hidden_states: NamedArray, layer_idx, inference: bool = True,
 If you're not used to the `tril`-as-a-mask trick, it's a way to create a causal mask so that the attention scores
 for a query can only attend to the past. The `tril` function creates a lower triangular matrix. It's equivalent to:
 ```python
-causal_mask = jnp.zeros(seq_len, key_seq_len)
+causal_mask = jnp.zeros((seq_len, key_seq_len))
 for i in range(seq_len):
     for j in range(key_seq_len):
-        if j >= i:
-            mask[i, j] = 1
+        if j <= i:
```

Comment:
Since it is lower triangular matrix, I believe the sign should be `<=` instead of `>=`.

With `>= i:`, we get: 

```
[[1. 1. 1.]
 [0. 1. 1.]
 [0. 0. 1.]]
```

Instead, with `<= i`, we get:
```
[[1. 0. 0.]
 [1. 1. 0.]
 [1. 1. 1.]]
```

Reply:
Actually, I found a critical difference in implementation with minGPT:

In minGPT ([here](https://github.com/karpathy/minGPT/blob/master/mingpt/model.py#L63)), it sets cells to be `-inf` in which the tril matrix (self.bias) is 0

```python
self.register_buffer("bias", torch.tril(torch.ones(config.block_size, config.block_size)).view(1, 1, config.block_size, config.block_size))
att = att.masked_fill(self.bias[:,:,:T,:T] == 0, float('-inf'))
```

Whereas in Levantar's implementation, it sets `-1e9` in places where causal_mask is 1. 

```
causal_mask = hax.tril(hax.ones((self.SeqLen, self.KeySeqLen), dtype=bool), self.SeqLen, self.KeySeqLen)
attn_scores = hax.where(causal_mask, attn_scores, -1e9)
```

What do you think @dlwh ? Did I miss anything in the implementation?

Reply:
Please skip my earlier comment on the comparison of tril between minGPT and levanter. I realized the nuance.